### ISSUE TITLE
ak.numbers_to_type is setting buffer itemsize incorrectly

### ISSUE THREAD

**Issue by jpivarski** [scikit-hep/awkward#383]
@ianna Minimal reproducer:

```python
>>> ak.numbers_to_type(ak.Array([{"x": 1.1}, {"x": 3.3}]), np.float32) + 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jpivarski/miniconda3/lib/python3.8/site-packages/numpy/lib/mixins.py", line 21, in func
    return ufunc(self, other)
  File "/home/jpivarski/irishep/awkward-1.0/awkward1/highlevel.py", line 1275, in __array_ufunc__
    return awkward1._connect._numpy.array_ufunc(ufunc, method, inputs, kwargs)
  File "/home/jpivarski/irishep/awkward-1.0/awkward1/_connect/_numpy.py", line 105, in array_ufunc
    out = awkward1._util.broadcast_and_apply(inputs, getfunction, behavior)
  File "/home/jpivarski/irishep/awkward-1.0/awkward1/_util.py", line 820, in broadcast_and_apply
    out = apply(broadcast_pack(inputs, isscalar), 0)
  File "/home/jpivarski/irishep/awkward-1.0/awkward1/_util.py", line 659, in apply
    outcontent = apply(nextinputs, depth + 1)
  File "/home/jpivarski/irishep/awkward-1.0/awkward1/_util.py", line 743, in apply
    apply(
  File "/home/jpivarski/irishep/awkward-1.0/awkward1/_util.py", line 487, in apply
    return function()
  File "/home/jpivarski/irishep/awkward-1.0/awkward1/_connect/_numpy.py", line 100, in <lambda>
    awkward1.layout.NumpyArray(getattr(ufunc, method)(*inputs, **kwargs)),
RuntimeError: Item size 8 for PEP 3118 buffer format string f does not match the dtype f item size 4.
```

Also, @henryiii suggested  better name: `ak.numbers_astype` (only for the high-level Python function; the low-level method can stay as-is).

**Comment by jpivarski** [comment]
This also happens if you change type from `float64` to `float64`. It doesn't have to be different.

**Comment by jpivarski** [comment]
Nevermind; fixed in #385.
